### ISSUE TITLE
[5.x][help needed] Fix autocomplete javascript

### ISSUE THREAD

**Issue by TomasVotruba** [ApiGen/ApiGen#881]
There was support for class, interface, trait, function and namespace autocomplete in search input.

## How it Should Work

1. Start typing a class name (fuzzy search will happen)
2. Pick your class by arrows or mouse
3. Hit "enter"
3. It will redirect to url

You can try it yourself in [Doctrne API](http://www.doctrine-project.org/api/orm/2.5/) or just watch me recoding it for you:

![aae](https://user-images.githubusercontent.com/924196/27001982-dc04c93e-4dd6-11e7-8c6e-5c67bc137d35.gif)

I've upgrade to new Jquery 3.1, changed format of original data provided by `elementlist.js` file and dropped lof of Javascript - this all broke it. I'm no Javascript master, so need help with this.

## Current `elementlist.js` format 

- element name => filepath to redirect

```javascript
var ApiGen = ApiGen || {};
ApiGen.elements = {
    "ApiGen\\Tests\\Generator\\Source":"namespace-ApiGen.Tests.Generator.Source.html",
    "ApiGen\\Tests\\Generator\\Source\\someFunction()":"function-ApiGen.Tests.Generator.Source.someFunction.html",
    "ApiGen\\Tests\\Generator\\Source\\SomeClass":"class-ApiGen.Tests.Generator.Source.SomeClass.html",
    "ApiGen\\Tests\\Generator\\Source\\SomeInterface":"interface-ApiGen.Tests.Generator.Source.SomeInterface.html",
    "ApiGen\\Tests\\Generator\\Source\\SomeTrait":"trait-ApiGen.Tests.Generator.Source.SomeTrait.html"   
};
```

## Important files with autocomplete source code

- [elementlist.js](https://github.com/ApiGen/ApiGen/blob/master/packages/ThemeDefault/src/elementlist.js.latte) - this is template for generating data in format mentioned above
- [resources/main.js](https://github.com/ApiGen/ApiGen/blob/8a7e1399532fdf0709910f6888fe6944f9054153/packages/ThemeDefault/src/resources/main.js#L8-L72) - javascript code for old autocomplete, I expect it to be much simpler now, as before URL was generated by javascript parsing and now it's the name in array.
- jsfiddle for Jquery 3.1 and JqueryUI with autocomplete - https://jsfiddle.net/jphellemons/0ukbtgs4/
- [resources/style.css](https://github.com/ApiGen/ApiGen/blob/8a7e1399532fdf0709910f6888fe6944f9054153/packages/ThemeDefault/src/resources/style.css#L72-L115) - style for autocomplete pop-up list

I admit I'm complete noob in javavascript, but I'll try to guide you as much as I can, so we can resolve this.

Thanks a lot for any help

cc @jankal @POPSuL

**Comment by vlastavesely** [comment]
It happened that in our company's system I need similar autocomplete. So, tomorrow I will start to work on this, probably it will be done before the return typehints #928

**Comment by POPSuL** [comment]
I solve this issue

**Comment by vlastavesely** [comment]
I already have it and now I am testing it.

**Comment by TomasVotruba** [comment]
I'm glad you're so active :+1: 
Feel free to send both of your PRs, I think we can work it out for the best.

**Comment by POPSuL** [comment]
@vlastavesely im too =(
![image](https://user-images.githubusercontent.com/683358/27912113-6e25379c-62a7-11e7-8e32-132412e46ece.png)

**Comment by TomasVotruba** [comment]
Resolved by #932
